core/primitives/src/types.rs

```diff
@@ -513,3 +514,44 @@ pub enum ValidatorKickoutReason {
 
 #[derive(PartialEq, Eq, Clone, Debug, BorshSerialize, BorshDeserialize, Serialize)]
 pub struct StateHeaderKey(pub ShardId, pub CryptoHash);
+
+/// Represents a rational number. This struct is used exclusively in configs so
+/// `u32` provides enough precision. Since we do not expect to have computation
+/// that returns a `Fraction`, there is no guarantee that it is an irreducible fraction.
+#[derive(Debug, Serialize, Deserialize, Hash, PartialEq, Eq, Clone, Copy)]
+pub struct Fraction {
+    pub numerator: u32,
+    pub denominator: u32,
+}
```

Comment:
I know this is a small implementation, but have you considered using an existing crates https://crates.io/crates/num-rational and https://crates.io/crates/fraction/? :thinking:

If we still want to maintain this implementation ourselves, I want to suggest we move it into its own module as this `types.rs` is already big enough, and I suggest we only define types that represent the core blockchain types here (`Fraction` is a generic type).

Reply:
We don't need most of the arithmetic operations so I prefer to have our own struct here.